### PR TITLE
Update `get-deps` Makefile targets

### PR DESCRIPTION
**Pull request by karuppiah7890**
## What this PR does / why we need it

_Note: This is a follow up of #2470 PR_

Most get-deps targets were using "go mod download", which isn't as
accurate as "go mod tidy" for evaluating dependencies. From one of the
go maintainers:

```
go mod download is downloading all of the modules in the dependency
graph, which it can determine from reading only the go.mod files. It
doesn't know which of those modules are actually needed to satisfy a
build, so it doesn't pull in the checksums for those modules (because
they may not be relevant).

On the other hand, go mod tidy has to walk the package graph in order
to ensure that all imports are satisfied. So it knows, for a fact,
without doing any extra work, that the modules it walks through are
needed for a go build in some configuration.
```

## Details for the Release Notes (PLEASE PROVIDE)

```release-note
NONE
```

## Which issue(s) this PR fixes

--

## Describe testing done for PR

--

## Special notes for your reviewer

This is a follow up of #2470 based on the conversation [here](https://github.com/vmware-tanzu/community-edition/pull/2470#discussion_r746874456). This is based on the commit from #2498 by @stmcginnis

### CODE REVIEW

**Comment by karuppiah7890** [comment]
TODOs
- Wait for #2470 to be merged
- Rebase with `main`

**Review by ankeesler — DISMISSED** [review]
pinniped change lgtm

**Review by stmcginnis — DISMISSED** [review]

**Review by jpmcb — APPROVED** [review]
This looks good to me from a CLI plugins build perspective!

On a new VM with Go installed, I did `make build-cli-plugins-local` and saw it download the necessary dependencies and produce the binaries.

**Comment by jpmcb** [comment]
Hmm looks like my review was for a number of groups. Would still be good to get approval from
- @vmware-tanzu/pkg-calico-owners 
- @vmware-tanzu/pkg-external-dns-owners 
- @vmware-tanzu/pkg-harbor-owners 
- @vmware-tanzu/pkg-multus-cni-owners 

and any others I missed

**Review by mcwumbly — APPROVED** [review]

**Comment by dvonthenen** [comment]
going to merge this since this has been in the queue for some time

**Pull request merged**
